### PR TITLE
Updating Shifts API Docs

### PR DESCRIPTION
**Pull request by worycatts**
Closes out [BH-74372](https://jira.bullhorn.com/browse/BH-74372)

Added API docs for shifts related entities and updated the Notes entity with new fields

Shift (shift definition)
ShiftPosition
JobShift
CandidateAvailability
CandidatePreferences
ShiftType
JobShiftSubmission
JobShiftAssignment
Note (add JobShifts as an entity reference)

### CODE REVIEW

File: source/includes/entityref/_candidateshiftpreference.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# CandidateShiftPreference
+
+CandidatePreferences is used for storing a candidate's preferences for times and days when they want to work in association with the shiftTypes that are configured.
```
Comment (Allblues) [inline source/includes/entityref/_candidateshiftpreference.md:3]: Change to something like:
Stores a candidate's preferences for times and days to work in association with the shiftTypes that are configured.

If ShiftType is an entity, use a capital S on it.

File: source/includes/entityref/_candidateshiftpreference.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            <td>id</td>
+            <td>Integer</td>
+            <td>Unique identifier for this entity.</td>
+            <td>X</td>
+            <td>X</td>
+        </tr>
+        <tr class="odd">
+            <td>candidate</td>
+            <td>To-one association</td>
+            <td>Candidate associated to Shift Preferences.</td>
```
Comment (Allblues) [inline source/includes/entityref/_candidateshiftpreference.md:33]: Here and elsewhere use the camelcased real entity name.

File: source/includes/entityref/_jobshift.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
+# JobShift
+
+JobShift is the main entity for shift scheduling and is used for storing start & end times, references to the clientCorporation, job, and other details for the shift (such as location, name, etc).
```
Comment (Allblues) [inline source/includes/entityref/_jobshift.md:3]: Change to something like:
The main entity for shift scheduling. Used for storing start and end times, references to the ClientCorporation, JobOrder, and other details for the shift such as location, name, and so forth.

Throughout the doc, use the real camelcased entity name where appropriate.

File: source/includes/entityref/_jobshift.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+            <td>endTime</td>
+            <td>OffsetDateTime</td>
+            <td>Time the Job Shift Ends.</td>
+            <td></td>
+            <td></td>
+        </tr>
+        <tr class="even">
+            <td>fillRatio</td>
+            <td>Double</td>
+            <td>Ratio of Assigned Candidates to Openings on the Job Shift.</td>
```
Comment (Allblues) [inline source/includes/entityref/_jobshift.md:54]: Change to:
Ratio of assigned Candidates to Openings on the JobShift.

What is Openings? Is it an actual entity or just a general thing? If general, use lowercase on it.

File: source/includes/entityref/_jobshift.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            <td>jobShiftAssignments</td>
+            <td>To-many association</td>
+            <td>Job Shift Assignments associated to the Job Shift.</td>
+            <td></td>
+            <td>X</td>
+        </tr>
+        <tr class="even">
+            <td>jobShiftSubmissions</td>
+            <td>To-many association</td>
+            <td>Job Shift Submission associated to a shift. Job Shift Submission is created when Offering a Candidate a Shift, before the Shift Submission is promoted to a Shift Assignment.</td>
```
Comment (Allblues) [inline source/includes/entityref/_jobshift.md:82]: Same comment about entity names as elsewhere. Seems "Offering" should be lowercase.

File: source/includes/entityref/_jobshift.md [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            <td>shiftPosition</td>
+            <td>To-one association</td>
+            <td>Shift Position associated with the Job Shift.</td>
+            <td></td>
+            <td></td>
+        </tr>
+        <tr class="even">
+            <td>shiftTypes</td>
+            <td>To-many association</td>
+            <td>The shift types the associated Job Shift spans over. Shift Types are designated in Shift Admin.</td>
```
Comment (Allblues) [inline source/includes/entityref/_jobshift.md:152]: Maybe try:
The ShiftTypes over which the associated JobShift spans.

Is Shift Admin an real application name?

File: source/includes/entityref/_jobshiftassignment.md [hunk 7]
```diff
@@ -0,0 +1,136 @@
+# JobShiftAssignment
+
+JobShiftAssignment ties the candidate to a JobShift and also allows for storing time capture details (such as actualStartTime and actualEndTime).
```
Comment (Allblues) [inline source/includes/entityref/_jobshiftassignment.md:3]: Change to:
Ties the Candidate to a JobShift and also allows for storing time capture details such as actualStartTime and actualEndTime.

File: source/includes/entityref/_shift.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+            <td>id</td>
+            <td>Integer</td>
+            <td>Unique identifier for this entity.</td>
+            <td>X</td>
+            <td>X</td>
+        </tr>
+        <tr class="odd">
+            <td>clientCorporation</td>
+            <td>To-one association</td>
+            <td>Client Corporation associated to the Shift Defintion.</td>
```
Comment (Allblues) [inline source/includes/entityref/_shift.md:33]: Fix typo  - Defintion should be Definition

File: source/includes/entityref/_shiftposition.md [hunk 9]
```diff
@@ -0,0 +1,227 @@
+# ShiftPosition
+
+ShiftPosition is used to create more specific shifts tied to a single job. Also can be used for storing rate information that is tied to the jobShift.
```
Comment (Allblues) [inline source/includes/entityref/_shiftposition.md:3]: Change to:
Used to create more specific Shifts tied to a single JobOrder. Also can be used for storing rate information that is tied to the JobShift.

I'm not sure if you want real entity names like used above.

File: source/includes/entityref/_shifttype.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
+# ShiftType
+
+ShiftType is an entity set at the corporation level to define various shift time intervals (Morning, Afternoon, Evening, Overnight, etc.) - used for setting candidatePreferences.
```
Comment (Allblues) [inline source/includes/entityref/_shifttype.md:3]: Try to avoid "etc" and instead use "and so forth".

**Review by Allblues — CHANGES_REQUESTED** [review]

File: source/includes/entityref/_candidateavailability.md [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+            <td>id</td>
+            <td>Integer</td>
+            <td>Unique identifier for this entity.</td>
+            <td>X</td>
+            <td>X</td>
+        </tr>
+        <tr class="odd">
+            <td>candidate</td>
+            <td>To-one association</td>
+            <td>Candidate associated to this availability.</td>
```
[the remark this answers is not in the document]
Comment (worycatts) [reply source/includes/entityref/_candidateavailability.md]: fixed.

**Pull request merged**